src/input/SimModSuite.h

```diff
@@ -84,78 +99,76 @@ class SimModSuite : public MeshInput
 
 		// Load CAD
 		logInfo(PMU_rank()) << "Loading model";
-		std::string sCadFile;
-		if (cadFile)
-			sCadFile = cadFile;
-		else {
-			sCadFile = modFile;
-			utils::StringUtils::replaceLast(sCadFile, ".smd", "_nat.x_t");
-		}
-		pNativeModel nativeModel = 0L;
-		if (utils::Path(sCadFile).exists())
-			nativeModel = ParasolidNM_createFromFile(sCadFile.c_str(), 0);
-
-		m_model = GM_load(modFile, nativeModel, 0L);
-		nativeModel = GM_nativeModel(m_model);
-
-		if (nativeModel)
-			NM_release(nativeModel);
-
-		// check for model errors
-		pPList modelErrors = PList_new();
-		if (!GM_isValid(m_model, 0, modelErrors))
-			// TODO print more detail about errors
-			logError() << "Input model is not valid";
-		PList_delete(modelErrors);
-
-		// Extract cases
-		logInfo(PMU_rank()) << "Extracting cases";
-		pAManager attMngr = SModel_attManager(m_model);
-
-		MeshingOptions meshingOptions;
-		pACase meshCaseFile = extractCase(attMngr, meshCaseName);
-		AttCase_associate(meshCaseFile, NULL);
-		pACase meshCase = MS_newMeshCase(m_model);
-		MS_processSimModelerMeshingAtts(meshCaseFile, meshCase, &meshingOptions);
-		AttCase_setModel(meshCase, m_model);
-
-		pACase analysisCase = extractCase(attMngr, analysisCaseName);
-		pPList children = AttNode_children(analysisCase);
-		void* iter = 0L;
-		while (pANode child = static_cast<pANode>(PList_next(children, &iter))) {
-			AttCase_setModel(static_cast<pACase>(child), m_model);
-    }
-		PList_delete(children);
 
-		if (nativeModel)
+      std::string smodFile = modFile;
+      if (cadFile!=0L) {
+         loadCAD(modFile, cadFile);
+      } else if(smodFile.substr(smodFile.find_last_of(".") + 1) == "smd") {
+         loadCAD(modFile, cadFile);
+      } else {
+		 logError() << "unsupported model file";
+      }
+
+      // Extract cases
+      logInfo(PMU_rank()) << "Extracting cases";
+      pACase meshCase, analysisCase;
+      MeshAttributes MeshAtt;
+
+      if(xmlFile != NULL) {
+
+        //Read mesh Attributes from xml file
+        int numFaces = GM_numFaces(m_model);
+        MeshAtt.init(xmlFile);
+        AnalysisAttributes AnalysisAtt(xmlFile, numFaces);
+        setCases(m_model, meshCase, analysisCase, MeshAtt, AnalysisAtt);
+      } else {
+        extractCases(m_model, meshCase, meshCaseName, analysisCase, analysisCaseName);
+      }
+
+		//if (nativeModel)
 			m_simMesh = PM_new(0, m_model, PMU_size());
-		else
+		//else
```

Comment:
Can be removed, right?

Reply:
About that, these changes were made by Johannes K., TUM bachelor student at the time.
I just built up the XML features on top of his branch. I mean it's not very clear to me why he did these changes.

Today, I checked what are the changes relative to the native models.
It turns out that these if else statements were because discrete models were not run in parallel by SImModeler at the time but native models were.

one line was removed without a clear goal:
https://github.com/SeisSol/PUMGen/commit/c6314a1bce816ebef2c70e27cfb47e0fe17ecb9d#diff-5b0a53ca5d510247569353bf859e6119bc05af5de8aaf1c0fd9a79ba7d74590fL98
I ve put it back. The code still compiles.

I think this should retrieve the compatibility with native model, but I don't have a file to test that.

Reply:
(see new commits)

Reply:
Okay, so let's leave it like this until someone wants to run with a native model.